Compiler bug when using a method on a redeclared []int type

### Describe the bug

Build fails to compile the included snippet.

### Expected Behavior

Build to succeed or to output a relevant error.

### Current Behavior

```
v -cg run main.v
/tmp/v_501/main.11590083687242138526.tmp.c:12500:29: error: use of undeclared identifier '_var_93'
        array_set(x, i, &(int[]) { _var_93 });
                                   ^
/tmp/v_501/main.11590083687242138526.tmp.c:12501:29: error: use of undeclared identifier '_var_87'
        array_set(x, j, &(int[]) { _var_87 });
                                   ^
2 errors generated.
builder error:
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

### Reproduction Steps

```
module main

pub type IntSlice = []int

pub fn (mut x IntSlice) swap(i int, j int) {
	x[i], x[j] = x[j], x[i]
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.0 59eb76c

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.0 b0323c7.59eb76c
OS: macos, macOS, 13.4.1, 22F82
Processor: 12 cpus, 64bit, little endian, Apple M2 Pro

getwd: /Users/alex/dev/vin
vexe: /opt/homebrew/Cellar/vlang/0.4/libexec/v
vexe mtime: 2023-07-09 15:34:55

vroot: OK, value: /opt/homebrew/Cellar/vlang/0.4/libexec
VMODULES: OK, value: /Users/alex/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.39.2 (Apple Git-143)
Git vroot status: weekly.2023.27-40-g59eb76c8
.git/config present: true

CC version: Apple clang version 14.0.3 (clang-1403.0.22.14.1)
thirdparty/tcc status: thirdparty-macos-arm64 a668e5a0
```

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -889,7 +889,7 @@ fn (mut g Gen) gen_cross_var_assign(node &ast.AssignStmt) {
 				}
 			}
 			ast.IndexExpr {
-				sym := g.table.sym(left.left_type)
+				sym := g.table.sym(g.table.unaliased_type(left.left_type))
 				if sym.kind == .array {
 					info := sym.info as ast.Array
 					elem_typ := g.table.sym(info.elem_type)

--- a/vlib/v/tests/cross_assign_aliased_array_test.v
+++ b/vlib/v/tests/cross_assign_aliased_array_test.v
@@ -1,0 +1,13 @@
+pub type IntSlice = []int
+
+pub fn (mut x IntSlice) swap(i int, j int) {
+	x[i], x[j] = x[j], x[i]
+}
+
+fn test_cross_assign_aliased_array() {
+	mut x := IntSlice([11, 22])
+	println(x)
+	x.swap(0, 1)
+	println(x)
+	assert x == IntSlice([22, 11])
+}

